includes/Plugin.php

```diff
@@ -219,31 +219,70 @@ public static function checkPluginCompatabilityAndPrintErrorMessages(): bool
      */
     public function validateSelectedFulfillmentStates(): void
     {
-        $fulfillmentState = $this->settingsRepository->getFulfillmentState();
-        $fulfilledState = $this->settingsRepository->getFulfilledState();
-        $fulfillmentErrorState = $this->settingsRepository->getFulfillmentErrorState();
+        $states = [
+            'Fulfillment State'       => $this->settingsRepository->getFulfillmentState(),
+            'Pending State'           => $this->settingsRepository->getPendingState(),
+            'Fulfilled State'         => $this->settingsRepository->getFulfilledState(),
+            'Fulfillment Error State' => $this->settingsRepository->getFulfillmentErrorState(),
+        ];
+
+        foreach ($states as $k => $v) {
+            $this->checkIfFulfillmentSettingExists($v, $k);
+        }
 
-        $this->checkIfFulfillmentSettingExists($fulfillmentState, "Fulfillment State");
-        $this->checkIfFulfillmentSettingExists($fulfilledState, "Fulfilled State");
-        $this->checkIfFulfillmentSettingExists($fulfillmentErrorState, "Fulfillment Error State");
+        $duplicateStates = $this->keysOfDuplicateValues($states);
+        if ($duplicateStates === null) {
+            $this->showWordpressAdminErrorMessage("keysOfDuplicateValues: provided array must only have strings");
+            return;
+        }
 
-        if ($fulfillmentState === $fulfilledState) {
-            $this->showWordpressAdminErrorMessage(
-                __("The Gebrüder Weiss WooCommerce Plugin settings for Fulfillment State and Fulfilled State are set to the same state.", self::LANGUAGE_DOMAIN)
-            );
+        if ($duplicateStates === []) {
```

Comment:
why the silent return here?

Reply:
i do too. i would just add a comment, like: no duplicate states found, everything below is error message generation.
For me it was not instantly clear why you could mark it as valid if no duplicate states were found.

Reply:
fixed